ui/pages/swaps/prepare-swap-page/review-quote.js

```diff
@@ -0,0 +1,1307 @@
+    switch (chainId) {
+      case CHAIN_IDS.MAINNET:
+        return t('networkNameEthereum');
+      case CHAIN_IDS.BSC:
+        return t('networkNameBSC');
+      case CHAIN_IDS.POLYGON:
+        return t('networkNamePolygon');
+      case CHAIN_IDS.LOCALHOST:
+        return t('networkNameTestnet');
+      case CHAIN_IDS.GOERLI:
+        return t('networkNameGoerli');
+      case CHAIN_IDS.AVALANCHE:
+        return t('networkNameAvalanche');
+      case CHAIN_IDS.OPTIMISM:
+        return t('networkNameOptimism');
+      case CHAIN_IDS.ARBITRUM:
+        return t('networkNameArbitrum');
+      default:
+        throw new Error('This network is not supported for token swaps');
+    }
+  };
+
+  let gasFeeInputs;
+  if (networkAndAccountSupports1559) {
+    // For Swaps we want to get 'high' estimations by default.
+    // eslint-disable-next-line react-hooks/rules-of-hooks
+    gasFeeInputs = useGasFeeInputs(GasRecommendations.high, {
+      userFeeLevel: swapsUserFeeLevel || GasRecommendations.high,
+    });
+  }
+
+  const fetchParamsSourceToken = fetchParams?.sourceToken;
+
+  const additionalTrackingParams = {
+    reg_tx_fee_in_usd: undefined,
+    reg_tx_fee_in_eth: undefined,
+    reg_tx_max_fee_in_usd: undefined,
+    reg_tx_max_fee_in_eth: undefined,
+    stx_fee_in_usd: undefined,
+    stx_fee_in_eth: undefined,
+    stx_max_fee_in_usd: undefined,
+    stx_max_fee_in_eth: undefined,
+  };
+
+  const usedGasLimit =
+    usedQuote?.gasEstimateWithRefund ||
+    `0x${decimalToHex(usedQuote?.averageGas || 0)}`;
+
+  const gasLimitForMax = usedQuote?.gasEstimate || `0x0`;
+
+  const usedGasLimitWithMultiplier = new BigNumber(gasLimitForMax, 16)
+    .times(usedQuote?.gasMultiplier || FALLBACK_GAS_MULTIPLIER, 10)
+    .round(0)
+    .toString(16);
+
+  const nonCustomMaxGasLimit = usedQuote?.gasEstimate
+    ? usedGasLimitWithMultiplier
+    : `0x${decimalToHex(usedQuote?.maxGas || 0)}`;
+  const maxGasLimit = customMaxGas || nonCustomMaxGasLimit;
+
+  let maxFeePerGas;
+  let maxPriorityFeePerGas;
+  let baseAndPriorityFeePerGas;
+
+  // EIP-1559 gas fees.
+  if (networkAndAccountSupports1559) {
+    const {
+      maxFeePerGas: suggestedMaxFeePerGas,
+      maxPriorityFeePerGas: suggestedMaxPriorityFeePerGas,
+      gasFeeEstimates: { estimatedBaseFee = '0' },
+    } = gasFeeInputs;
+    maxFeePerGas = customMaxFeePerGas || decGWEIToHexWEI(suggestedMaxFeePerGas);
+    maxPriorityFeePerGas =
+      customMaxPriorityFeePerGas ||
+      decGWEIToHexWEI(suggestedMaxPriorityFeePerGas);
+    baseAndPriorityFeePerGas = addHexes(
+      decGWEIToHexWEI(estimatedBaseFee),
+      maxPriorityFeePerGas,
+    );
+  }
+  let gasTotalInWeiHex = calcGasTotal(maxGasLimit, maxFeePerGas || gasPrice);
+  if (multiLayerL1FeeTotal !== null) {
+    gasTotalInWeiHex = sumHexes(
+      gasTotalInWeiHex || '0x0',
+      multiLayerL1FeeTotal || '0x0',
+    );
+  }
+
+  const { tokensWithBalances } = useTokenTracker(swapsTokens, true);
```

Comment:
Did you check the edge case we mentioned on slack, where the user selects a token to swap that they have not yet added to their token list? The possible failure, given that this PR removes the `destinationTokenAddedForSwap` and associated logic, is that token balances are not calculated correctly on this page for destination tokens that are not in the users list.